docs/zkapps/tutorials/11-advanced-account-updates.mdx

```diff
@@ -0,0 +1,169 @@
+  const myTokenInstance = new MyToken(myTokenAddr);
+  const tokenUserInstance = new TokenUser(tokenUserAddr);
+  const tokenHolderInstance = new TokenHolder(tokenUserAddr, myTokenInstance.token.id);
+```
+
+And deployed as follows:
+
+```typescript
+  const deploy_txn = await Mina.transaction(deployerAccount, () => {
+    let feePayerUpdate = AccountUpdate.fundNewAccount(deployerAccount, 4);
+    feePayerUpdate.send({ to: myTokenAddr, amount: accountFee });
+
+    myTokenInstance.deploy();
+    tokenUserInstance.deploy();
+    tokenHolderInstance.deploy()
+
+    myTokenInstance.approveDeploy(tokenHolderInstance.self);
+  });
+```
+
+The special thing to note here is the `approveDeploy(...)` on the `myTokenInstance`. Because `tokenHolderInstance` is going to exist as a token account, it needs approval to be deployed.
+
+The `myTokenInstance.approveDeploy(...)` code is as follows:
+
+```typescript
+  @method approveDeploy(deployUpdate: AccountUpdate) {
+    // TODO should anything else be asserted here for a "safe" deployment?
+    this.approve(deployUpdate, AccountUpdate.Layout.NoChildren);
+
+    // see if balance change is zero
+    let balanceChange = Int64.fromObject(deployUpdate.body.balanceChange);
+    balanceChange.assertEquals(Int64.from(0));
+  }
+```
+
+This checks that the deployUpdate is a single account update, with no children - and that it isn't creating any additional tokens, and is starting out with a balance of zero.
+
+Next, here is an example of sending tokens from the `tokenUser`/`tokenHolder` to another account:
+
+```typescript
+  const txn2 = await Mina.transaction(deployerAccount, () => {
+    let feePayerUpdate = AccountUpdate.fundNewAccount(deployerAccount, 1);
+    tokenUserInstance.sendMyTokens(UInt64.from(100), deployerAccount.toPublicKey());
+  });
+```
+
+The `fundNewAccount` call is included as the deployerAccount in this case doesn't have an existing token account.
+
+`tokenUserInstance.sendMyTokens(...)` is implemented as follows:
+
+
+```typescript
+  @method sendMyTokens(
+    amount: UInt64, 
+    destination: PublicKey
+  ) {
+    const tokenHolder = this.tokenHolder;
+    tokenHolder.transfer(destination, amount);
+    this.tokenContract.approveTransfer(tokenHolder.self);
+  }
+```
+
+Which calls the `transfer` contract on `tokenHolder`:
+
+```typescript
+  @method transfer(to: PublicKey, amount: UInt64) {
+    const toAccountUpdate = AccountUpdate.create(to, this.tokenId);
+    toAccountUpdate.balance.addInPlace(amount);
+    toAccountUpdate.body.mayUseToken =
+      AccountUpdate.MayUseToken.InheritFromParent;
+
+    this.balance.subInPlace(amount);
+
+    this.self.body.mayUseToken = AccountUpdate.MayUseToken.ParentsOwnToken;
+  }
+```
+
+And is approved by the token contract's `approveTransfer`:
+
+```typescript
+  @method approveTransfer(
+    transferUpdate: AccountUpdate
+  ) {
+    // TODO should anything else be asserted here for a "safe" deployment?
+    this.approve(transferUpdate);
```

Comment:
the default for `this.approve()` is to require the child to have no children. so this shouldn't work when passed the `transfer` account update, which creates a child as the receiver. The only pattern that works and is safe to my knowledge is the "approveUpdateAndSend" thing in the WrappedMina example

Reply:
yes 👍🏻